x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts

```diff
@@ -194,6 +195,7 @@ export interface BulkActionResult {
   updated: Rule[];
   created: Rule[];
   deleted: Rule[];
+  skipped: Rule[];
```

Comment:
I think it should be an object with some rule attributes rather than a whole rule.

Because we skip one or many bulk actions in general, we could end up with many reasons why a rule could be skipped. For instance, a rule could be skipped because it doesn't exist, if we decide to treat ids of not found rules as non-errors.

Action point I'd suggest: let's define the interfaces for the response of the bulk action endpoint under `x-pack/plugins/security_solution/common/detection_engine/rule_management/api/rules/bulk_actions`, and specify it as follows:

```ts
export interface BulkActionResult {
  ...
  skipped: BulkActionSkipResult[];
}

export interface BulkActionSkipResult {
  id: RuleObjectId;
  name?: RuleName;
  reason_code: BulkActionSkipReason;
}

export enum BulkActionSkipReason {
  FIRST_REASON = 'FIRST_REASON', // replace with actual reasons
  SECOND_REASON = 'SECOND_REASON',
}
```

We're going to have similar skip results in the new rule upgrade and installation endpoints (https://github.com/elastic/kibana/pull/144060), so I'd like to make it consistent across the whole `detection_engine` API.

Reply:
Created `x-pack/plugins/security_solution/common/detection_engine/rule_management/api/rules/bulk_actions/response_schema.ts` with response types.